In [2]:
import pandas as pd
import numpy as np

## Data load

In [3]:
# 데이터 불러오기
kp20 = pd.read_csv('_1_accident analysis code csv파일//KP2020.csv', encoding='cp949')
kp21 = pd.read_csv('_1_accident analysis code csv파일/KP2021.csv', encoding='cp949')
npa20 = pd.read_csv('_1_accident analysis code csv파일//NPA2020.csv', encoding='cp949')

## Data merge

In [4]:
# EVT_CL_CD가 401(교통사고)인 행 추출
kp2020 = kp20[kp20['EVT_CL_CD']==401]
kp2021 = kp21[kp21['EVT_CL_CD']==401]
npa2020 = npa20[npa20['EVT_CL_CD']==401]

In [5]:
# kp2020 RECV_CPLT_DM 컬럼 값 변경 (뒤에 0000000 제거)
split1 = kp2020.RECV_CPLT_DM.str.split('.')
a_datetime = split1.str.get(0)
kp2020['RECV_CPLT_DM'] = a_datetime
kp2020

C:\Users\passion\AppData\Local\Temp\ipykernel_25864\2313164009.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kp2020['RECV_CPLT_DM'] = a_datetime


RECV_DEPT_NM       RECV_CPLT_DM  NPA_CL  EVT_STAT_CD  EVT_CL_CD  \
4              충남청  20/12/01 08:17:50      19           10        401   
9              대전청  20/12/01 07:08:44      13           10        401   
10             대전청  20/12/01 07:09:20      13           10        401   
31             충남청  20/12/01 01:36:38      19           10        401   
53             대전청  20/12/01 01:39:03      13           10        401   
...            ...                ...     ...          ...        ...   
76995          충남청  20/12/31 00:33:54      19           10        401   
76998          충남청  20/12/31 00:34:45      19           10        401   
77036          대전청  20/12/31 01:08:43      13           10        401   
77068          대전청  20/12/31 01:17:01      13           10        401   
77069          대전청  20/12/31 01:18:53      13           10        401   

       RPTER_SEX                   HPPN_PNU_ADDR      HPPN_X     HPPN_Y  \
4            2.0  충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160  36.826718   
9            1.0         대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724   
10           2.0      대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694   
31           1.0  충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290  36.967612   
53           1.0       대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597   
...          ...                             ...         ...        ...   
76995        1.0        충청남도 서산시 온석동(동문1동) 354-2  126.471991  36.801294   
76998        1.0        충청남도 서산시 온석동(동문1동) 393-3  126.471756  36.801603   
77036        1.0           대전광역시 중구 대흥동(행정:대흥동)   127.428343  36.324386   
77068        1.0     대전광역시 서구 도마동(행정:도마2동) 549-1  127.369099  36.312971   
77069        1.0        대전광역시 서구 도마동(도마2동) 549-1  127.368945  36.312984   

      SME_EVT_YN  
4            NaN  
9            NaN  
10           NaN  
31           NaN  
53           NaN  
...          ...  
76995          Y  
76998          Y  
77036        NaN  
77068        NaN  
77069        NaN  

[6207 rows x 10 columns]

In [6]:
# kp2021 RECV_CPLT_DM 컬럼 값 변경 (뒤에 0000000 제거)
split2 = kp2021.RECV_CPLT_DM.str.split('.')
b_datetime = split2.str.get(0)
kp2021['RECV_CPLT_DM'] = b_datetime
kp2021

C:\Users\passion\AppData\Local\Temp\ipykernel_25864\520476093.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kp2021['RECV_CPLT_DM'] = b_datetime


RECV_DEPT_NM       RECV_CPLT_DM  NPA_CL  EVT_STAT_CD  EVT_CL_CD  \
23               대전청  21/03/07 00:07:54      13           10        401   
26               대전청  21/03/07 00:15:27      13           10        401   
49               충남청  21/03/07 00:13:58      19           10        401   
109              대전청  21/03/07 02:17:35      13           10        401   
157              충남청  21/01/03 00:12:25      19           10        401   
...              ...                ...     ...          ...        ...   
2593987          대전청  23/01/14 17:04:36      13           10        401   
2593994          대전청  23/01/15 20:01:39      13           10        401   
2594002          충남청  23/01/16 21:26:56      19           10        401   
2594019          대전청  23/01/14 17:35:38      13           10        401   
2594035          대전청  23/01/15 01:13:20      13           10        401   

         RPTER_SEX                    HPPN_PNU_ADDR      HPPN_X     HPPN_Y  \
23             2.0         대전광역시 서구 괴정동(괴정동) 367-40  127.371854  36.341961   
26             1.0              대전광역시 유성구 상대동 471-2  127.339334  36.346899   
49             1.0  충청남도 천안시 동남구 목천읍 운전리(목천읍) 333-1  127.230796  36.767596   
109            2.0        대전광역시 서구 갈마동(갈마2동) 343-30  127.370973  36.347592   
157            2.0       충청남도 천안시 서북구 두정동(부성2동) 633  127.130635  36.825850   
...            ...                              ...         ...        ...   
2593987        2.0                    대전 동구 원동 51-1  127.433184  36.329560   
2593994        1.0                   대전 서구 괴정동 88-6  127.384284  36.334507   
2594002        1.0     충청남도 태안군 근흥면 두야리(근흥면 ) 246-2  126.262393  36.751467   
2594019        1.0  대전광역시 대덕구 신탄진동 (행정:신탄진동 ) 129-7  127.433785  36.450725   
2594035        1.0      대전광역시 대덕구 평촌동 (행정:덕암동 ) 346  127.428287  36.425403   

        SME_EVT_YN  
23             NaN  
26             NaN  
49             NaN  
109            NaN  
157            NaN  
...            ...  
2593987        NaN  
2593994        NaN  
2594002        NaN  
2594019        NaN  
2594035          Y  

[164582 rows x 10 columns]

In [7]:
#kp2021 kp2020 데이터 합치기
kp_merged = pd.concat([kp2020,kp2021], axis=0)
kp_merged.reset_index(inplace=True)
kp_merged = kp_merged.drop(['index'], axis=1)
kp_merged

RECV_DEPT_NM       RECV_CPLT_DM  NPA_CL  EVT_STAT_CD  EVT_CL_CD  \
0               충남청  20/12/01 08:17:50      19           10        401   
1               대전청  20/12/01 07:08:44      13           10        401   
2               대전청  20/12/01 07:09:20      13           10        401   
3               충남청  20/12/01 01:36:38      19           10        401   
4               대전청  20/12/01 01:39:03      13           10        401   
...             ...                ...     ...          ...        ...   
170784          대전청  23/01/14 17:04:36      13           10        401   
170785          대전청  23/01/15 20:01:39      13           10        401   
170786          충남청  23/01/16 21:26:56      19           10        401   
170787          대전청  23/01/14 17:35:38      13           10        401   
170788          대전청  23/01/15 01:13:20      13           10        401   

        RPTER_SEX                    HPPN_PNU_ADDR      HPPN_X     HPPN_Y  \
0             2.0   충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160  36.826718   
1             1.0          대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724   
2             2.0       대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694   
3             1.0   충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290  36.967612   
4             1.0        대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597   
...           ...                              ...         ...        ...   
170784        2.0                    대전 동구 원동 51-1  127.433184  36.329560   
170785        1.0                   대전 서구 괴정동 88-6  127.384284  36.334507   
170786        1.0     충청남도 태안군 근흥면 두야리(근흥면 ) 246-2  126.262393  36.751467   
170787        1.0  대전광역시 대덕구 신탄진동 (행정:신탄진동 ) 129-7  127.433785  36.450725   
170788        1.0      대전광역시 대덕구 평촌동 (행정:덕암동 ) 346  127.428287  36.425403   

       SME_EVT_YN  
0             NaN  
1             NaN  
2             NaN  
3             NaN  
4             NaN  
...           ...  
170784        NaN  
170785        NaN  
170786        NaN  
170787        NaN  
170788          Y  

[170789 rows x 10 columns]

In [8]:
#kp_merged RECV_CPLT_DM컬럼으로 RECV_date, RECV_time split 컬럼 생성
split3 = kp_merged.RECV_CPLT_DM.str.split(' ')
a_date = split3.str.get(0)
a_time = split3.str.get(1)
kp_merged['RECV_date'],kp_merged['RECV_time'] = (a_date,a_time)

In [9]:
#RECV_date컬럼 year,month,day로 split
date_split = kp_merged.RECV_date.str.split('/')
a_year = date_split.str.get(0)
a_month = date_split.str.get(1)
a_day = date_split.str.get(2)
kp_merged['RECV_year'],kp_merged['RECV_month'],kp_merged['RECV_day'] = (a_year,a_month,a_day)
kp_merged = kp_merged.drop(['RECV_date'], axis=1)

In [10]:
#RECV_time컬럼 hour,minute,second로 split
time_split = kp_merged.RECV_time.str.split(':')
a_hour = time_split.str.get(0)
a_minute = time_split.str.get(1)
a_second = time_split.str.get(2)
kp_merged['RECV_hour'],kp_merged['RECV_minute'],kp_merged['RECV_second'] = (a_hour,a_minute,a_second)
kp_merged = kp_merged.drop(['RECV_time'], axis=1)

In [11]:
kp_merged

RECV_DEPT_NM       RECV_CPLT_DM  NPA_CL  EVT_STAT_CD  EVT_CL_CD  \
0               충남청  20/12/01 08:17:50      19           10        401   
1               대전청  20/12/01 07:08:44      13           10        401   
2               대전청  20/12/01 07:09:20      13           10        401   
3               충남청  20/12/01 01:36:38      19           10        401   
4               대전청  20/12/01 01:39:03      13           10        401   
...             ...                ...     ...          ...        ...   
170784          대전청  23/01/14 17:04:36      13           10        401   
170785          대전청  23/01/15 20:01:39      13           10        401   
170786          충남청  23/01/16 21:26:56      19           10        401   
170787          대전청  23/01/14 17:35:38      13           10        401   
170788          대전청  23/01/15 01:13:20      13           10        401   

        RPTER_SEX                    HPPN_PNU_ADDR      HPPN_X     HPPN_Y  \
0             2.0   충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160  36.826718   
1             1.0          대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724   
2             2.0       대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694   
3             1.0   충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290  36.967612   
4             1.0        대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597   
...           ...                              ...         ...        ...   
170784        2.0                    대전 동구 원동 51-1  127.433184  36.329560   
170785        1.0                   대전 서구 괴정동 88-6  127.384284  36.334507   
170786        1.0     충청남도 태안군 근흥면 두야리(근흥면 ) 246-2  126.262393  36.751467   
170787        1.0  대전광역시 대덕구 신탄진동 (행정:신탄진동 ) 129-7  127.433785  36.450725   
170788        1.0      대전광역시 대덕구 평촌동 (행정:덕암동 ) 346  127.428287  36.425403   

       SME_EVT_YN RECV_year RECV_month RECV_day RECV_hour RECV_minute  \
0             NaN        20         12       01        08          17   
1             NaN        20         12       01        07          08   
2             NaN        20         12       01        07          09   
3             NaN        20         12       01        01          36   
4             NaN        20         12       01        01          39   
...           ...       ...        ...      ...       ...         ...   
170784        NaN        23         01       14        17          04   
170785        NaN        23         01       15        20          01   
170786        NaN        23         01       16        21          26   
170787        NaN        23         01       14        17          35   
170788          Y        23         01       15        01          13   

       RECV_second  
0               50  
1               44  
2               20  
3               38  
4               03  
...            ...  
170784          36  
170785          39  
170786          56  
170787          38  
170788          20  

[170789 rows x 16 columns]

In [12]:
npa2020

RECV_CPLT_DT  RECV_CPLT_TM  NPA_CL  EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
54           20200101          2135      13           10        401         1   
66           20200101          4301      13           10        401         1   
117          20200101        184335      13           10        401         3   
144          20200101        190853      13           10        401         1   
147          20200101        191525      13           10        401         1   
...               ...           ...     ...          ...        ...       ...   
1177991      20201120        125706      13           10        401         1   
1178035      20201122           322      13            5        401         1   
1178051      20201122          5410      13            5        401         1   
1178086      20201122          4008      13            5        401         1   
1178173      20201122          3753      13            5        401         1   

                         HPPN_OLD_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  
54       대전광역시 유성구 봉명동(행정:온천1동) 541-26  127.348625  36.353829        NaN  
66         대전광역시 중구 오류동(행정:오류동) 187-27  127.405055  36.328573        NaN  
117                   대전광역시 동구 가양동 630  127.445780  36.345500          N  
144          대전광역시 동구 판암동(행정:판암2동) 237  127.459160  36.323490        NaN  
147           대전광역시 중구 태평동(태평2동) 378-7  127.394136  36.326646        NaN  
...                                ...         ...        ...        ...  
1177991      대전광역시 유성구 도룡동(행정:신성동) 3-1  127.384612  36.377690          Y  
1178035   대전광역시 유성구 봉명동(행정:온천1동) 651-1  127.347209  36.357588             
1178051   대전광역시 유성구 장대동(행정:온천2동) 375-3  127.339844  36.358391             
1178086       대전광역시 동구 신흥동(행정:신인동) 212  127.446090  36.320140          Y  
1178173       대전광역시 동구 신흥동(행정:신인동) 212  127.446090  36.320140          Y  

[93659 rows x 10 columns]

In [13]:
#npa20020 HPPN_OLD_ADDR 컬럼 명칭 변경
npa2020 = npa2020.rename(columns={'HPPN_OLD_ADDR':'HPPN_PNU_ADDR'})

In [14]:
#npa2020의 날짜와 시간 컬럼 kp_merged와 형식 맞추기
npa2020['RECV_CPLT_TM'] = npa2020["RECV_CPLT_TM"].astype('str').str.zfill(6)
npa2020['RECV_CPLT_DT'] = npa2020["RECV_CPLT_DT"].astype('str')
npa2020['RECV_CPLT_DT'] = npa2020["RECV_CPLT_DT"].apply(lambda x : x[2:])
npa2020.head()

RECV_CPLT_DT RECV_CPLT_TM  NPA_CL  EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
54        200101       002135      13           10        401         1   
66        200101       004301      13           10        401         1   
117       200101       184335      13           10        401         3   
144       200101       190853      13           10        401         1   
147       200101       191525      13           10        401         1   

                     HPPN_PNU_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  
54   대전광역시 유성구 봉명동(행정:온천1동) 541-26  127.348625  36.353829        NaN  
66     대전광역시 중구 오류동(행정:오류동) 187-27  127.405055  36.328573        NaN  
117               대전광역시 동구 가양동 630  127.445780  36.345500          N  
144      대전광역시 동구 판암동(행정:판암2동) 237  127.459160  36.323490        NaN  
147       대전광역시 중구 태평동(태평2동) 378-7  127.394136  36.326646        NaN

In [15]:
#npa2020 year / month / day / hour / minute / second 컬럼 생성
npa2020['RECV_year'] = npa2020["RECV_CPLT_DT"].apply(lambda x : x[0:2])
npa2020['RECV_month'] = npa2020["RECV_CPLT_DT"].apply(lambda x : x[2:4])
npa2020['RECV_day'] = npa2020["RECV_CPLT_DT"].apply(lambda x : x[4:])
npa2020['RECV_hour'] = npa2020["RECV_CPLT_TM"].apply(lambda x : x[0:2])
npa2020['RECV_minute'] = npa2020["RECV_CPLT_TM"].apply(lambda x : x[2:4])
npa2020['RECV_second'] = npa2020["RECV_CPLT_TM"].apply(lambda x : x[4:])
npa2020.head()

RECV_CPLT_DT RECV_CPLT_TM  NPA_CL  EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
54        200101       002135      13           10        401         1   
66        200101       004301      13           10        401         1   
117       200101       184335      13           10        401         3   
144       200101       190853      13           10        401         1   
147       200101       191525      13           10        401         1   

                     HPPN_PNU_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  \
54   대전광역시 유성구 봉명동(행정:온천1동) 541-26  127.348625  36.353829        NaN   
66     대전광역시 중구 오류동(행정:오류동) 187-27  127.405055  36.328573        NaN   
117               대전광역시 동구 가양동 630  127.445780  36.345500          N   
144      대전광역시 동구 판암동(행정:판암2동) 237  127.459160  36.323490        NaN   
147       대전광역시 중구 태평동(태평2동) 378-7  127.394136  36.326646        NaN   

    RECV_year RECV_month RECV_day RECV_hour RECV_minute RECV_second  
54         20         01       01        00          21          35  
66         20         01       01        00          43          01  
117        20         01       01        18          43          35  
144        20         01       01        19          08          53  
147        20         01       01        19          15          25

In [16]:
#npa2020 RECV_CPLT_DM 컬럼 생성 및 RECV_CPLT_DT / RECV_CPLT_TM 컬럼 제거
npa2020['RECV_year'] = npa2020["RECV_year"].astype('str')
npa2020['RECV_month'] = npa2020["RECV_month"].astype('str')
npa2020['RECV_day'] = npa2020["RECV_day"].astype('str')
npa2020['RECV_hour'] = npa2020["RECV_hour"].astype('str')
npa2020['RECV_minute'] = npa2020["RECV_minute"].astype('str')
npa2020['RECV_second'] = npa2020["RECV_second"].astype('str')
npa2020['RECV_CPLT_DM'] = npa2020['RECV_year']+'/'+npa2020['RECV_month']+'/'+npa2020['RECV_day']+' '+npa2020['RECV_hour']+':'+npa2020['RECV_minute']+':'+npa2020['RECV_second']
npa2020 = npa2020.drop(['RECV_CPLT_DT'], axis=1)
npa2020 = npa2020.drop(['RECV_CPLT_TM'], axis=1)

In [17]:
#불필요한 컬럼 RECV_DEPT_NM 제거
kp_merged = kp_merged.drop(['RECV_DEPT_NM'], axis=1)

In [18]:
#kp_merged 와 npa2020 데이터 합치기
accident = pd.concat([kp_merged,npa2020], axis=0)
accident

RECV_CPLT_DM  NPA_CL  EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
0        20/12/01 08:17:50      19           10        401       2.0   
1        20/12/01 07:08:44      13           10        401       1.0   
2        20/12/01 07:09:20      13           10        401       2.0   
3        20/12/01 01:36:38      19           10        401       1.0   
4        20/12/01 01:39:03      13           10        401       1.0   
...                    ...     ...          ...        ...       ...   
1177991  20/11/20 12:57:06      13           10        401         1   
1178035  20/11/22 00:03:22      13            5        401         1   
1178051  20/11/22 00:54:10      13            5        401         1   
1178086  20/11/22 00:40:08      13            5        401         1   
1178173  20/11/22 00:37:53      13            5        401         1   

                          HPPN_PNU_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  \
0        충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160  36.826718        NaN   
1               대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724        NaN   
2            대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694        NaN   
3        충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290  36.967612        NaN   
4             대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597        NaN   
...                                 ...         ...        ...        ...   
1177991       대전광역시 유성구 도룡동(행정:신성동) 3-1  127.384612  36.377690          Y   
1178035    대전광역시 유성구 봉명동(행정:온천1동) 651-1  127.347209  36.357588              
1178051    대전광역시 유성구 장대동(행정:온천2동) 375-3  127.339844  36.358391              
1178086        대전광역시 동구 신흥동(행정:신인동) 212  127.446090  36.320140          Y   
1178173        대전광역시 동구 신흥동(행정:신인동) 212  127.446090  36.320140          Y   

        RECV_year RECV_month RECV_day RECV_hour RECV_minute RECV_second  
0              20         12       01        08          17          50  
1              20         12       01        07          08          44  
2              20         12       01        07          09          20  
3              20         12       01        01          36          38  
4              20         12       01        01          39          03  
...           ...        ...      ...       ...         ...         ...  
1177991        20         11       20        12          57          06  
1178035        20         11       22        00          03          22  
1178051        20         11       22        00          54          10  
1178086        20         11       22        00          40          08  
1178173        20         11       22        00          37          53  

[264448 rows x 15 columns]

#### 3개의 raw data를 1개의 dataframe <span style="color:blue;">'accident'</span>로 합쳤다. 

## Data preprocessing

In [19]:
#결측치 확인
accident.isnull().sum()

RECV_CPLT_DM          0
NPA_CL                0
EVT_STAT_CD           0
EVT_CL_CD             0
RPTER_SEX             0
HPPN_PNU_ADDR      7466
HPPN_X             2770
HPPN_Y             2770
SME_EVT_YN       176531
RECV_year             0
RECV_month            0
RECV_day              0
RECV_hour             0
RECV_minute           0
RECV_second           0
dtype: int64

In [20]:
accident['HPPN_Y'].value_counts()

36.341685    6228
0.000000     2366
36.817772     114
36.833200     108
36.351133     101
             ... 
36.352345       1
36.915092       1
36.822264       1
36.673677       1
36.358391       1
Name: HPPN_Y, Length: 162158, dtype: int64

In [21]:
accident['HPPN_X'].value_counts()

127.404663    6228
0.000000      2366
127.108062     116
127.131963     108
127.119179     101
              ... 
127.342498       1
127.379375       1
127.346921       1
127.422026       1
127.339844       1
Name: HPPN_X, Length: 165314, dtype: int64

HPPN_X와 HPPN_Y 컬럼의 데이터가 2770개의 null 값을 가지고 있는 것을 확인했고, 값이 0으로 된 데이터가 2366개 있는것을 확인했다.<br>
위도와 경도가 없는 데이터는 구할 수 없어 삭제했다.

In [22]:
#위도,경도 null값 0으로 대체
accident['HPPN_Y'] = accident['HPPN_Y'].fillna(0)
accident['HPPN_X'] = accident['HPPN_X'].fillna(0)

In [23]:
#위도,경도 값이 0인 데이터 삭제
accident = accident[accident.HPPN_X != 0]
accident = accident[accident.HPPN_Y != 0]

In [24]:
accident

RECV_CPLT_DM  NPA_CL  EVT_STAT_CD  EVT_CL_CD RPTER_SEX  \
0        20/12/01 08:17:50      19           10        401       2.0   
1        20/12/01 07:08:44      13           10        401       1.0   
2        20/12/01 07:09:20      13           10        401       2.0   
3        20/12/01 01:36:38      19           10        401       1.0   
4        20/12/01 01:39:03      13           10        401       1.0   
...                    ...     ...          ...        ...       ...   
1177991  20/11/20 12:57:06      13           10        401         1   
1178035  20/11/22 00:03:22      13            5        401         1   
1178051  20/11/22 00:54:10      13            5        401         1   
1178086  20/11/22 00:40:08      13            5        401         1   
1178173  20/11/22 00:37:53      13            5        401         1   

                          HPPN_PNU_ADDR      HPPN_X     HPPN_Y SME_EVT_YN  \
0        충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160  36.826718        NaN   
1               대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724        NaN   
2            대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694        NaN   
3        충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290  36.967612        NaN   
4             대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597        NaN   
...                                 ...         ...        ...        ...   
1177991       대전광역시 유성구 도룡동(행정:신성동) 3-1  127.384612  36.377690          Y   
1178035    대전광역시 유성구 봉명동(행정:온천1동) 651-1  127.347209  36.357588              
1178051    대전광역시 유성구 장대동(행정:온천2동) 375-3  127.339844  36.358391              
1178086        대전광역시 동구 신흥동(행정:신인동) 212  127.446090  36.320140          Y   
1178173        대전광역시 동구 신흥동(행정:신인동) 212  127.446090  36.320140          Y   

        RECV_year RECV_month RECV_day RECV_hour RECV_minute RECV_second  
0              20         12       01        08          17          50  
1              20         12       01        07          08          44  
2              20         12       01        07          09          20  
3              20         12       01        01          36          38  
4              20         12       01        01          39          03  
...           ...        ...      ...       ...         ...         ...  
1177991        20         11       20        12          57          06  
1178035        20         11       22        00          03          22  
1178051        20         11       22        00          54          10  
1178086        20         11       22        00          40          08  
1178173        20         11       22        00          37          53  

[259312 rows x 15 columns]

Null 값 2770개와 값이 0인 데이터 2366개가 제거되어 264,448개의 데이터에서 259,312개의 데이터가 된 것을 확인할 수 있다.

In [25]:
accident['SME_EVT_YN'].value_counts()

Y    47090
     26403
N    12822
Name: SME_EVT_YN, dtype: int64

In [26]:
#재신고여부 결측치 처리
accident['SME_EVT_YN'] = accident['SME_EVT_YN'].fillna('N')
accident=accident.replace({'SME_EVT_YN':' '},'N')

In [27]:
accident['SME_EVT_YN'].value_counts()

N    212222
Y     47090
Name: SME_EVT_YN, dtype: int64

In [28]:
#불필요한 컬럼 제거
accident = accident.drop(['NPA_CL'], axis=1)      #경찰청 구분
accident = accident.drop(['EVT_STAT_CD'], axis=1) #사건상태코드
accident = accident.drop(['RPTER_SEX'], axis=1)   #신고자 성별
accident = accident.drop(['SME_EVT_YN'], axis=1)  #동일사건여부
accident = accident.drop(['RECV_minute'], axis=1)  #minute
accident = accident.drop(['RECV_second'], axis=1)  #seocnd
accident = accident.drop(['EVT_CL_CD'], axis=1)  #사건종결코드
accident

RECV_CPLT_DM                   HPPN_PNU_ADDR      HPPN_X  \
0        20/12/01 08:17:50  충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160   
1        20/12/01 07:08:44         대전광역시 중구 석교동(석교동) 85-14  127.447545   
2        20/12/01 07:09:20      대전광역시 중구 석교동(행정:석교동) 88-11  127.447175   
3        20/12/01 01:36:38  충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290   
4        20/12/01 01:39:03       대전광역시 동구 대동(행정:대동) 355-14  127.447390   
...                    ...                             ...         ...   
1177991  20/11/20 12:57:06       대전광역시 유성구 도룡동(행정:신성동) 3-1  127.384612   
1178035  20/11/22 00:03:22    대전광역시 유성구 봉명동(행정:온천1동) 651-1  127.347209   
1178051  20/11/22 00:54:10    대전광역시 유성구 장대동(행정:온천2동) 375-3  127.339844   
1178086  20/11/22 00:40:08        대전광역시 동구 신흥동(행정:신인동) 212  127.446090   
1178173  20/11/22 00:37:53        대전광역시 동구 신흥동(행정:신인동) 212  127.446090   

            HPPN_Y RECV_year RECV_month RECV_day RECV_hour  
0        36.826718        20         12       01        08  
1        36.306724        20         12       01        07  
2        36.306694        20         12       01        07  
3        36.967612        20         12       01        01  
4        36.324597        20         12       01        01  
...            ...       ...        ...      ...       ...  
1177991  36.377690        20         11       20        12  
1178035  36.357588        20         11       22        00  
1178051  36.358391        20         11       22        00  
1178086  36.320140        20         11       22        00  
1178173  36.320140        20         11       22        00  

[259312 rows x 8 columns]

In [29]:
#accident 데이터 index 초기화
accident.reset_index(inplace=True)
accident = accident.drop(['index'], axis=1)
accident

RECV_CPLT_DM                   HPPN_PNU_ADDR      HPPN_X  \
0       20/12/01 08:17:50  충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160   
1       20/12/01 07:08:44         대전광역시 중구 석교동(석교동) 85-14  127.447545   
2       20/12/01 07:09:20      대전광역시 중구 석교동(행정:석교동) 88-11  127.447175   
3       20/12/01 01:36:38  충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290   
4       20/12/01 01:39:03       대전광역시 동구 대동(행정:대동) 355-14  127.447390   
...                   ...                             ...         ...   
259307  20/11/20 12:57:06       대전광역시 유성구 도룡동(행정:신성동) 3-1  127.384612   
259308  20/11/22 00:03:22    대전광역시 유성구 봉명동(행정:온천1동) 651-1  127.347209   
259309  20/11/22 00:54:10    대전광역시 유성구 장대동(행정:온천2동) 375-3  127.339844   
259310  20/11/22 00:40:08        대전광역시 동구 신흥동(행정:신인동) 212  127.446090   
259311  20/11/22 00:37:53        대전광역시 동구 신흥동(행정:신인동) 212  127.446090   

           HPPN_Y RECV_year RECV_month RECV_day RECV_hour  
0       36.826718        20         12       01        08  
1       36.306724        20         12       01        07  
2       36.306694        20         12       01        07  
3       36.967612        20         12       01        01  
4       36.324597        20         12       01        01  
...           ...       ...        ...      ...       ...  
259307  36.377690        20         11       20        12  
259308  36.357588        20         11       22        00  
259309  36.358391        20         11       22        00  
259310  36.320140        20         11       22        00  
259311  36.320140        20         11       22        00  

[259312 rows x 8 columns]

In [30]:
#accident의 복사본 → HPPN_PNU_ADDR컬럼의 이상한 주소와 NULL 값 뽑아내기 위한 dataframe
strange=accident

In [31]:
#HPPN_PNU_ADDR컬럼의 null과 이상한 데이터 뽑아내기
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('세종',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('대전',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('충청남도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('충남',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('충청북도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('경기도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('전라북도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('서울',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('전라남도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('제주',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('대구',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('부산',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('경상북도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('경상남도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('광주',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('강원도',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('인천',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('경기',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('울산',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('충북',na=False)==False]
strange = strange[strange['HPPN_PNU_ADDR'].str.contains('전북 ',na=False)==False]

In [32]:
strange['HPPN_PNU_ADDR'].value_counts()

                             702
복지회관 앞 주차장                     1
소원면 송현3리1014-3                 1
경부선 상행                         1
연서면(행정:연서면)                    1
미래빌딩 앞 사거리                     1
천안방향                           1
완주군 봉동읍 제내리                    1
경부/죽암휴게소7키로전/하행                1
경부/하행선/천안에서청주내려가는/             1
요                              1
부강면(행정:부강면)                    1
고덕중앙로 65-1                     1
대로변에서 자연아파트 들어가는 사차선도로         1
Read timed out                 1
서천군 판교면 판교리 407, (서천사랑병원)      1
[이관주소] 중구-신창동3가--:2:           1
(행정:연기면)                       1
Name: HPPN_PNU_ADDR, dtype: int64

In [33]:
#HPPN_PNU_ADDR 결측치 처리
strange['HPPN_PNU_ADDR'] = strange['HPPN_PNU_ADDR'].fillna('Nan')
strange=strange.replace({'HPPN_PNU_ADDR':' '},'Nan')

In [34]:
#strange 데이터 index 재설정
strange.reset_index(inplace=True)
strange = strange.drop(['index'], axis=1)

In [35]:
#kakao map API를 사용하기 위한 csv파일 내보내기
strange.to_csv('_1_accident analysis code csv파일//KAKAO_map_API//strange.csv',mode="w",encoding='utf-8',index=False)

In [36]:
#kakao map API로 주소를 입력한 csv파일 불러오기
strange_solve = pd.read_csv('_1_accident analysis code csv파일//KAKAO_map_API//strange_solve.csv',encoding='utf-8')
strange_solve

0
0     대전광역시 중구 중촌동
1     대전광역시 중구 중촌동
2     대전광역시 중구 중촌동
3     대전광역시 중구 중촌동
4     대전광역시 중구 중촌동
...            ...
6304  대전광역시 중구 중촌동
6305  대전광역시 중구 중촌동
6306  대전광역시 중구 중촌동
6307  대전광역시 중구 중촌동
6308  대전광역시 중구 중촌동

[6309 rows x 1 columns]

In [39]:
#strange와 strange_solve 합치기
null_address = pd.concat([strange,strange_solve], axis=1)
null_address.head()

RECV_CPLT_DM HPPN_PNU_ADDR      HPPN_X     HPPN_Y RECV_year  \
0  20/12/01 14:28:47           Nan  127.404663  36.341685        20   
1  20/12/01 14:45:13           Nan  127.404663  36.341685        20   
2  20/12/01 14:24:06           Nan  127.404663  36.341685        20   
3  20/12/01 17:18:26           Nan  127.404663  36.341685        20   
4  20/12/01 17:30:42           Nan  127.404663  36.341685        20   

  RECV_month RECV_day RECV_hour             0  
0         12       01        14  대전광역시 중구 중촌동  
1         12       01        14  대전광역시 중구 중촌동  
2         12       01        14  대전광역시 중구 중촌동  
3         12       01        17  대전광역시 중구 중촌동  
4         12       01        17  대전광역시 중구 중촌동

In [40]:
#컬럼 삭제 및 컬럼 이름 변경
null_address = null_address.drop(['HPPN_PNU_ADDR'], axis=1)
null_address = null_address.rename(columns={'0':'HPPN_PNU_ADDR'})
null_address.head()

RECV_CPLT_DM      HPPN_X     HPPN_Y RECV_year RECV_month RECV_day  \
0  20/12/01 14:28:47  127.404663  36.341685        20         12       01   
1  20/12/01 14:45:13  127.404663  36.341685        20         12       01   
2  20/12/01 14:24:06  127.404663  36.341685        20         12       01   
3  20/12/01 17:18:26  127.404663  36.341685        20         12       01   
4  20/12/01 17:30:42  127.404663  36.341685        20         12       01   

  RECV_hour HPPN_PNU_ADDR  
0        14  대전광역시 중구 중촌동  
1        14  대전광역시 중구 중촌동  
2        14  대전광역시 중구 중촌동  
3        17  대전광역시 중구 중촌동  
4        17  대전광역시 중구 중촌동

In [41]:
null_address['HPPN_PNU_ADDR'].value_counts()

대전광역시 중구 중촌동            6208
충청남도 태안군 근흥면 도황리          51
대전광역시 동구 삼성동               3
대전광역시 서구 갈마동               2
충청북도 청주시 상당구 남일면 송암리       2
대전광역시 서구 둔산동               2
충청남도 천안시 서북구 불당동           2
충청남도 천안시 동남구 신부동           2
충청남도 천안시 서북구 차암동           2
대전광역시 유성구 용계동              1
세종특별자치시 도담동                1
대전광역시 유성구 송강동              1
대전광역시 유성구 궁동               1
대전광역시 유성구 구암동              1
충청남도 예산군 고덕면 대천리           1
세종특별자치시 부강면 금호리            1
충청남도 서산시 예천동               1
대전광역시 대덕구 읍내동              1
대전광역시 유성구 상대동              1
충청남도 태안군 소원면 송현리           1
전라북도 익산시 왕궁면 흥암리           1
충청남도 공주시 정안면 사현리           1
충청남도 아산시 탕정면 명암리           1
세종특별자치시 연서면 쌍전리            1
경기도 용인시 처인구 남사읍 봉명리        1
충청남도 천안시 서북구 성정동           1
대전광역시 서구 도안동               1
대전광역시 서구 봉곡동               1
대전광역시 유성구 노은동              1
충청남도 서천군 판교면 판교리           1
대전광역시 대덕구 중리동              1
전라남도 담양군 대전면 대치리           1
대전광역시 대덕구 오정동              1
대전광역시 중구 선화동               1
대전광역시 대덕구 법동  

주소가 없거나 이상했던 데이터들을 kakao mpa API를 이용하여 주소를 새로 불러왔다.<br>
여기에 대전,세종,충남에 속하지 않는 데이터는 제거하겠다.

In [42]:
#대전,충남,세종 이외의 데이터는 삭제
null_address = null_address[null_address['HPPN_PNU_ADDR'].str.contains('경기도',na=False)==False]
null_address = null_address[null_address['HPPN_PNU_ADDR'].str.contains('서울특별시',na=False)==False]
null_address = null_address[null_address['HPPN_PNU_ADDR'].str.contains('전라남도',na=False)==False]
null_address = null_address[null_address['HPPN_PNU_ADDR'].str.contains('충청북도',na=False)==False]
null_address = null_address[null_address['HPPN_PNU_ADDR'].str.contains('전라북도',na=False)==False]

## 최종데이터

In [43]:
#accident의 HPPN_PNU_ADDR 결측치 처리
accident['HPPN_PNU_ADDR'] = accident['HPPN_PNU_ADDR'].fillna('Nan')
accident=accident.replace({'HPPN_PNU_ADDR':' '},'Nan')

In [44]:
#accident와 null_address 합치기
city_all = pd.concat([accident,null_address], axis=0)
city_all

RECV_CPLT_DM                   HPPN_PNU_ADDR      HPPN_X  \
0     20/12/01 08:17:50  충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160   
1     20/12/01 07:08:44         대전광역시 중구 석교동(석교동) 85-14  127.447545   
2     20/12/01 07:09:20      대전광역시 중구 석교동(행정:석교동) 88-11  127.447175   
3     20/12/01 01:36:38  충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290   
4     20/12/01 01:39:03       대전광역시 동구 대동(행정:대동) 355-14  127.447390   
...                 ...                             ...         ...   
6304  20/11/20 09:01:14                    대전광역시 중구 중촌동  127.404663   
6305  20/11/20 11:09:05                    대전광역시 중구 중촌동  127.404663   
6306  20/11/20 14:47:06                    대전광역시 중구 중촌동  127.404663   
6307  20/11/20 17:59:14                    대전광역시 중구 중촌동  127.404663   
6308  20/11/20 09:55:10                    대전광역시 중구 중촌동  127.404663   

         HPPN_Y RECV_year RECV_month RECV_day RECV_hour  
0     36.826718        20         12       01        08  
1     36.306724        20         12       01        07  
2     36.306694        20         12       01        07  
3     36.967612        20         12       01        01  
4     36.324597        20         12       01        01  
...         ...       ...        ...      ...       ...  
6304  36.341685        20         11       20        09  
6305  36.341685        20         11       20        11  
6306  36.341685        20         11       20        14  
6307  36.341685        20         11       20        17  
6308  36.341685        20         11       20        09  

[265615 rows x 8 columns]

In [45]:
#중복되는 데이터 제거 후 index 초기화
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('충청북도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('경기도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('전라북도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('서울',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('전라남도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('제주',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('대구',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('부산',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('경상북도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('경상남도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('광주',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('강원도',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('인천',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('경기',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('울산',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('충북',na=False)==False]
city_all = city_all[city_all['HPPN_PNU_ADDR'].str.contains('전북 ',na=False)==False]
city_all = city_all[city_all.HPPN_PNU_ADDR != 'N']
city_all.reset_index(inplace=True)
city_all = city_all.drop(['index'], axis=1)

In [46]:
#지역을 구분짓기 위해 city 컬럼 생성
add_split = city_all.HPPN_PNU_ADDR.str.split(' ')
city = add_split.str.get(0)
city_all['city'] = city
city_all.head()

RECV_CPLT_DM                   HPPN_PNU_ADDR      HPPN_X     HPPN_Y  \
0  20/12/01 08:17:50  충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160  36.826718   
1  20/12/01 07:08:44         대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724   
2  20/12/01 07:09:20      대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694   
3  20/12/01 01:36:38  충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290  36.967612   
4  20/12/01 01:39:03       대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597   

  RECV_year RECV_month RECV_day RECV_hour   city  
0        20         12       01        08   충청남도  
1        20         12       01        07  대전광역시  
2        20         12       01        07  대전광역시  
3        20         12       01        01   충청남도  
4        20         12       01        01  대전광역시

In [47]:
city_all['city'].value_counts()

충청남도                  137492
대전광역시                 101818
세종특별자치시                13420
Nan                     6292
대전                      2962
충남                       366
[이관주소]충청남도                15
[이관주소]                    13
[공중전화]                     4
연서면(행정:연서면)                1
경부/하행선/천안에서청주내려가는/         1
경부선                        1
천안방향                       1
전원꺼짐대전광역시                  1
완주군                        1
경부/죽암휴게소7키로전/하행            1
소원면                        1
미래빌딩                       1
010-대전광역시                  1
부강면(행정:부강면)                1
고덕중앙로                      1
대로변에서                      1
[이관주소]대전광역시                1
Read                       1
서천군                        1
요                          1
복지회관                       1
(행정:연기면)                   1
Name: city, dtype: int64

위에 5개를 충청남도 / 대전광역시 / 세종특별자치시 / 대전 / 충남을 제외하고는<br>
모두 kakao map API를 통해 다시 주소를 입력해주었기에 중복되는 값으로 무시해도 된다.

In [48]:
#키워드 별로 지역 data 생성
city_all=city_all.replace({'city':'대전'},'대전광역시')
city_all=city_all.replace({'city':'충남'},'충청남도')
daejeon = city_all[city_all['city']=='대전광역시']
chungnam = city_all[city_all['city']=='충청남도']
sejong = city_all[city_all['city']=='세종특별자치시']

In [50]:
#index 초기화
daejeon.reset_index(inplace=True)
daejeon = daejeon.drop(['index','city'], axis=1)
chungnam.reset_index(inplace=True)
chungnam = chungnam.drop(['index','city'], axis=1)
sejong.reset_index(inplace=True)
sejong = sejong.drop(['index','city'], axis=1)

In [51]:
daejeon

RECV_CPLT_DM               HPPN_PNU_ADDR      HPPN_X     HPPN_Y  \
0       20/12/01 07:08:44     대전광역시 중구 석교동(석교동) 85-14  127.447545  36.306724   
1       20/12/01 07:09:20  대전광역시 중구 석교동(행정:석교동) 88-11  127.447175  36.306694   
2       20/12/01 01:39:03   대전광역시 동구 대동(행정:대동) 355-14  127.447390  36.324597   
3       20/12/01 00:12:30    대전광역시 유성구 교촌동(진잠동) 411-1  127.308197  36.308033   
4       20/12/01 00:09:52    대전광역시 유성구 교촌동(진잠동) 411-1  127.308197  36.308033   
...                   ...                         ...         ...        ...   
104775  20/11/20 09:01:14                대전광역시 중구 중촌동  127.404663  36.341685   
104776  20/11/20 11:09:05                대전광역시 중구 중촌동  127.404663  36.341685   
104777  20/11/20 14:47:06                대전광역시 중구 중촌동  127.404663  36.341685   
104778  20/11/20 17:59:14                대전광역시 중구 중촌동  127.404663  36.341685   
104779  20/11/20 09:55:10                대전광역시 중구 중촌동  127.404663  36.341685   

       RECV_year RECV_month RECV_day RECV_hour  
0             20         12       01        07  
1             20         12       01        07  
2             20         12       01        01  
3             20         12       01        00  
4             20         12       01        00  
...          ...        ...      ...       ...  
104775        20         11       20        09  
104776        20         11       20        11  
104777        20         11       20        14  
104778        20         11       20        17  
104779        20         11       20        09  

[104780 rows x 8 columns]

In [52]:
chungnam

RECV_CPLT_DM                   HPPN_PNU_ADDR      HPPN_X  \
0       20/12/01 08:17:50  충청남도 천안시 서북구 성정동(행정:성정2동) 1259  127.137160   
1       20/12/01 01:36:38  충청남도 서산시 대산읍 대로리(행정:대산읍) 216-3  126.412290   
2       20/12/01 02:23:03  충청남도 아산시 둔포면 송용리(행정:둔포면) 71-68  127.032318   
3       20/12/01 02:24:26         충청남도 서산시 읍내동(부춘동) 733-2  126.438445   
4       20/12/01 01:37:37     충청남도 서산시 대산읍 대로리(대산읍) 216-3  126.412245   
...                   ...                             ...         ...   
137853  20/08/18 13:58:20                충청남도 태안군 근흥면 도황리  126.191890   
137854  20/08/23 19:19:32                충청남도 태안군 근흥면 도황리  126.191890   
137855  20/09/01 10:18:15                충청남도 태안군 근흥면 도황리  126.191890   
137856  20/10/04 00:00:12                충청남도 태안군 근흥면 도황리  126.191890   
137857  20/10/24 18:50:12                충청남도 태안군 근흥면 도황리  126.191890   

           HPPN_Y RECV_year RECV_month RECV_day RECV_hour  
0       36.826718        20         12       01        08  
1       36.967612        20         12       01        01  
2       36.921510        20         12       01        02  
3       36.779188        20         12       01        02  
4       36.967579        20         12       01        01  
...           ...       ...        ...      ...       ...  
137853  36.690215        20         08       18        13  
137854  36.690215        20         08       23        19  
137855  36.690215        20         09       01        10  
137856  36.690215        20         10       04        00  
137857  36.690215        20         10       24        18  

[137858 rows x 8 columns]

In [53]:
sejong

RECV_CPLT_DM                   HPPN_PNU_ADDR      HPPN_X  \
0      20/12/01 13:22:45             세종특별자치시 조치원읍 원리 13   127.303180   
1      20/12/01 10:45:45  세종특별자치시 세종시 조치원읍 상리(조치원읍) 39-6  127.304611   
2      20/12/01 13:07:38    세종특별자치시 세종시 부강면 부강리(행정:부강면)   127.368902   
3      20/12/01 12:21:09    세종특별자치시 세종시 소정면 고등리(행정:소정면)   127.194694   
4      20/12/01 16:51:57     세종특별자치시 세종시 종촌동(행정:종촌동) 677  127.246969   
...                  ...                             ...         ...   
13415  20/11/20 10:12:32        세종특별자치시 세종시 어진동(행정:도담동)   127.267495   
13416  20/02/24 13:18:01                 세종특별자치시 부강면 금호리  127.364927   
13417  20/04/03 21:09:28                     세종특별자치시 도담동  127.258602   
13418  20/09/15 16:24:11                 세종특별자치시 연서면 쌍전리  127.276436   
13419  20/10/23 09:00:51                     세종특별자치시 해밀동  127.265246   

          HPPN_Y RECV_year RECV_month RECV_day RECV_hour  
0      36.601510        20         12       01        13  
1      36.603134        20         12       01        10  
2      36.522225        20         12       01        13  
3      36.717396        20         12       01        12  
4      36.504865        20         12       01        16  
...          ...       ...        ...      ...       ...  
13415  36.503282        20         11       20        10  
13416  36.491494        20         02       24        13  
13417  36.514305        20         04       03        21  
13418  36.572830        20         09       15        16  
13419  36.526412        20         10       23        09  

[13420 rows x 8 columns]

In [54]:
#RECV_CPLT_DM컬럼 datetime으로 변환
from datetime import datetime
daejeon['RECV_CPLT_DM']='20'+daejeon['RECV_CPLT_DM']
daejeon.RECV_CPLT_DM = daejeon.RECV_CPLT_DM.apply(lambda x: datetime.strptime(x, '%Y/%m/%d %H:%M:%S'))
daejeon['RECV_CPLT_DM'] = pd.to_datetime(daejeon['RECV_CPLT_DM'])
chungnam['RECV_CPLT_DM']='20'+chungnam['RECV_CPLT_DM']
chungnam.RECV_CPLT_DM = chungnam.RECV_CPLT_DM.apply(lambda x: datetime.strptime(x, '%Y/%m/%d %H:%M:%S'))
chungnam['RECV_CPLT_DM'] = pd.to_datetime(chungnam['RECV_CPLT_DM'])
sejong['RECV_CPLT_DM']='20'+sejong['RECV_CPLT_DM']
sejong.RECV_CPLT_DM = sejong.RECV_CPLT_DM.apply(lambda x: datetime.strptime(x, '%Y/%m/%d %H:%M:%S'))
sejong['RECV_CPLT_DM'] = pd.to_datetime(sejong['RECV_CPLT_DM'])

In [55]:
#csv 파일로 내보내기
daejeon.to_csv("_1_accident analysis code csv파일//daejeon.csv",mode="w",encoding='utf-8',index=False)
chungnam.to_csv("_1_accident analysis code csv파일//chungnam.csv",mode="w",encoding='utf-8',index=False)
sejong.to_csv("_1_accident analysis code csv파일//sejong.csv",mode="w",encoding='utf-8',index=False)